pkg/reconciler/knativeserving/common/ha.go

```diff
@@ -62,9 +62,26 @@ func HighAvailabilityTransform(instance *servingv1alpha1.KnativeServing, log *za
 			}
 		}
 
+		replicas := int64(instance.Spec.HighAvailability.Replicas)
+
 		// Transform deployments that support HA.
 		if u.GetKind() == "Deployment" && deploymentNames.Has(u.GetName()) {
-			if err := unstructured.SetNestedField(u.Object, int64(instance.Spec.HighAvailability.Replicas), "spec", "replicas"); err != nil {
+			if err := unstructured.SetNestedField(u.Object, replicas, "spec", "replicas"); err != nil {
+				return err
+			}
+		}
+
+		if u.GetKind() == "HorizontalPodAutoscaler" {
```

Comment:
Will all HPAs installed necessarily be ones which need to support HA?

Reply:
"An HPA being present is enough of a signal for us to assume that the component supports a HA configuration."

Sorry for missing this! Sounds reasonable to me.